gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -238,11 +246,45 @@ else if (statement instanceof ExecuteImmediate executeImmediate) {
         }
     }
 
-    private Map<String, String> getPreparedStatements(HttpServletRequest request)
+    private void processRequestBody(ContainerRequestContext requestContext)
+    {
+        if (!requestContext.hasEntity()) {
+            return;
+        }
+
+        MediaType mediaType = requestContext.getMediaType();
+        if (mediaType == null) {
+            return;
+        }
+
+        String charset = mediaType.getParameters().get("charset");
+        if (!StandardCharsets.UTF_8.name().equalsIgnoreCase(charset)) {
+            return;
```

Comment:
should we log a warning that it is missing charset? because request without charset is quite common isn't it?